app/controllers/application_controller.rb

```diff
@@ -44,8 +44,12 @@ def routing_error
       when Common::Exceptions::BaseError
         exception
       when Common::Client::Errors::BackendServiceError
-        additional_attributes = { source: exception.developer_message, meta: exception.as_json }
-        Common::Exceptions::ClientError.new(exception.message, additional_attributes)
+        additional_attributes = {
+          detail: exception.message,
+          source: exception.developer_message,
+          meta: exception.as_json
+        }
+        Common::Exceptions::BackendServiceError.new(nil, additional_attributes)
```

Comment:
I find it a bit confusing that we're converting a `BackendServiceError` to a `BackendServiceError`. Different namespaces, of course, but I don't think that will necessarily be obvious to future devs.